test/Microsoft.AspNetCore.Antiforgery.Test/Internal/DefaultClaimUidExtractorTest.cs

```diff
@@ -97,18 +99,80 @@ public void DefaultUniqueClaimTypes_NotPresent_SerializesAllClaimTypes()
         public void DefaultUniqueClaimTypes_Present()
         {
             // Arrange
-            var identity = new ClaimsIdentity();
+            var identity = new ClaimsIdentity("someAuthentication");
             identity.AddClaim(new Claim("fooClaim", "fooClaimValue"));
             identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, "nameIdentifierValue"));
 
             // Act
-            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(identity);
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(new ClaimsIdentity[] { identity });
 
             // Assert
             Assert.Equal(new string[]
             {
                 ClaimTypes.NameIdentifier,
                 "nameIdentifierValue",
+                "LOCAL AUTHORITY",
+            }, uniqueIdentifierParameters);
+        }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_PrefersSubClaimOverNameIdentifierAndUpn()
```

Comment:
Need to see more tests with unauthenticated identities, and with multiple authenticated identities for the ultimate fallback case.